components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,464 @@
+	-- see if match should actually be finished if score is set
+	if isScoreSet and autoFinished and not Logic.readBool(args.finished) then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', args.date))
+		local threshold = args.dateexact and 30800 or 86400
+		if matchUnixTime + threshold < currentUnixTime then
+			args.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	local winner = tostring(args.winner or '')
+	if Logic.readBool(args.finished) then
+		local placement = 1
+		local lastScore
+		local lastPlacement = 1
+		local lastStatus
+		-- luacheck: push ignore
+		for opponentIndex, opponent in Table.iter.spairs(opponents, CustomMatchGroupInput._placementSortFunction) do
+			if opponent.status ~= STATUS_HAS_SCORE and opponent.status ~= STATUS_DEFAULT_WIN and placement == 1 then
+				placement = 2
+			end
+			if placement == 1 then
+				args.winner = opponentIndex
+			end
+			if opponent.status == STATUS_HAS_SCORE and opponent.score == lastScore then
+				opponent.placement = lastPlacement
+			elseif opponent.status ~= STATUS_HAS_SCORE and opponent.status == lastStatus then
+				opponent.placement = lastPlacement
+			else
+				opponent.placement = placement
+			end
+			args['opponent' .. opponentIndex] = opponent
+			placement = placement + 1
+			lastScore = opponent.score
+			lastPlacement = opponent.placement
+			lastStatus = opponent.status
+		end
+	-- luacheck: pop
+	-- only apply arg changes otherwise
+	else
+		for opponentIndex, opponent in pairs(opponents) do
+			args['opponent' .. opponentIndex] = opponent
+		end
+	end
+	if
+		winner == RESULT_TYPE_DRAW or
+		winner == '0' or (
+			Logic.readBool(args.finished) and
+			#opponents == 2 and
+			opponents[1].status == STATUS_HAS_SCORE and
+			opponents[2].status == STATUS_HAS_SCORE and
+			opponents[1].score == opponents[2].score
+		)
+	then
+		args.winner = 0
+		args.resulttype = RESULT_TYPE_DRAW
+	elseif
+		Logic.readBool(args.finished) and
+		#opponents == MAX_NUM_OPPONENTS and
+		opponents[1].status ~= STATUS_HAS_SCORE and
+		opponents[1].status == opponents[2].status
+	then
+		args.winner = 0
+	end
+	return args
+end
+
+function matchFunctions.getPlayers(match, opponentIndex, teamName)
+	for playerIndex = 1, MAX_NUM_PLAYERS do
+		-- parse player
+		local player = Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex]) or {}
+		player.name = player.name or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.flag = player.flag or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'flag')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. playerIndex] = player
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+function mapFunctions.getExtraData(map)
+	local overtimes = Array.extractValues(Table.mapValues(mw.text.split(map.overtime or '', ','), tonumber))
```

Comment:
What's overtime in Trackmania?

Reply:
It's when two teams are equal to ten points on a map, it starts "Overtime", where it continues going to a point where a team has more point than another

Reply:
If it's ok for you we can resolve this conversation